apps/web/pages/api/cancel.ts

```diff
@@ -168,12 +173,17 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
   // by cancelling first, and blocking whilst doing so; we can ensure a cancel
   // action always succeeds even if subsequent integrations fail cancellation.
-  if (bookingToDelete.eventType?.recurringEvent) {
+  if (bookingToDelete.eventType?.recurringEvent && allRemainingBookings === "true") {
     const recurringEventId = bookingToDelete.recurringEventId;
     const where = recurringEventId === null ? { uid } : { recurringEventId };
-    // Proceed to mark as cancelled all recurring event instances
+    // Proceed to mark as cancelled all remaining recurring events instances (greater than or equal to right now)
     await prisma.booking.updateMany({
-      where,
+      where: {
+        ...where,
+        startTime: {
+          gte: new Date(),
```

Comment:
We just want to cancel the remaining recurring events, not past instances that were probably accepted may have happened.